src/context/context.tsx

```diff
@@ -0,0 +1,42 @@
+import React, { createContext, ReactNode, useEffect, useReducer } from 'react';
+import { usePosition } from 'hooks/usePosition';
+import { reducer } from './reducer';
+import { CHANGE_POSITION } from './actions';
+
+export const initState = {
+  position: {
+    latitude: 0,
+    longitude: 0
+  },
+  positionError: '',
+  changePosition: (lat: number, los: number) => {}
+};
+
+export const dataContext = createContext(initState);
+
+export const ContextProvider = ({ children }: { children: ReactNode }) => {
+  const { position, error } = usePosition();
+
+  const { Provider } = dataContext;
+
+  const [state, dispatch] = useReducer(reducer, initState);
+
+  state.changePosition = (lat, los) => {
```

Comment:
state should not be mutable. you don't need "changePosition" function inside state at all, make it a separate function and pass with context value